java/com/engflow/notificationqueue/Client.java

```diff
@@ -0,0 +1,207 @@
+package java.com.engflow.notificationqueue;
+
+import com.engflow.eventstore.v1.BuildLifecycleEventNotification;
+import com.engflow.eventstore.v1.EventStoreGrpc;
+import com.engflow.eventstore.v1.GetInvocationRequest;
+import com.engflow.eventstore.v1.StreamedBuildEvent;
+import com.engflow.notification.v1.*;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.grpc.ManagedChannel;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import io.netty.handler.ssl.SslContextBuilder;
+
+import javax.annotation.Nullable;
+import javax.net.ssl.SSLException;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+class Client {
+
+  public static void main(String[] args) throws Exception {
+    /**
+     * Check for the command line arguments in specific order.
+     * In this case, only the presence of the flags are checked.
+     * Create appropriated validations when needed.
+     */
+    if (!areAllInputsProvided(args)){
+      System.err.println("Please add the following flags: ");
+      System.err.println(" --notification_queue_endpoint");
+      System.err.println(" --queue_name");
+      System.err.println(" --tls_certificate");
+      System.err.println(" --tls_key");
+      return;
+    }
+
+    /**
+     * Channel used by the NotificationQueueGrpc and the EventStoreGrpc stub. It is also possible to
+     * use two separated channels.
+     */
+    ManagedChannel channel = null;
+    try {
+      channel =
```

Comment:
Let's split this into two `try` statements so we don't need that `if` statement inside `finally`. I also think `createChannel` should throw `IOException` so we can handle that here.

```
ManagedChannel channel = null;
try {
  channel = createChannel(...)
} catch (IOException e) {
  ...
}
try {
  pull(channel, args[1]);
} finally {
  channel.shutdownNow();
  ...
}
```